u3d/move command, renaming install dirs. Support long version names. Fixes #274

lacostej opened the pull request:
Fixes #274 

I am splitting this PR in various pieces. This PR is based on top:

[MERGED]  #290 #291 #292 #293 #294 

#295 #296

niezbop commented on lib/u3d_core/admin_tools.rb:
```diff
@@ -22,6 +22,17 @@
+    def self.move_os_file(os, source_path, new_path, dry_run:)
+      if os == :windows
+        source_path.tr!('/', '\\')
+        new_path.tr!('/', '\\')
```
Move `U3d::Utils.windows_path` to `U3dCore::Helper` and use it instead? It is a bit better than what you're using currently as it makes sure that specific paths will not cause issues (paths with numbers in IIRC).

niezbop commented:
Most of it looks good to me! It's a bit hard to see the difference with all those PR right now, it will probably be easier to get a grasp of the modifications once this PR is rebased on the other.

Aside from the windows path translation, that I would like to see changed (let's not keep bug prone behaviours if we are aware they can break), I think that this is good to go.

As a side note, I believe that this PR is big enough to qualify for at least a minor version number change, and almost a major if we consider the sanitation part of the PR that is being restricted to list, although it could be seen as a bug.

lacostej commented:
I rebased on top of master after merging all the other PRs. Gosh I love git.

I extracted the first 2 patches into #295 which you can review. I think we can add your change as a separate PR as it changes functionality, vs just refactoring now. WDYT?

niezbop commented:
Git FTW :)

Agreed for the separate PR. I'm not on my computer so I can't do this now though
